framework/src/utils/SlepcSupport.C

```diff
@@ -0,0 +1,257 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifndef SLEPCSUPPORT_H
+#define SLEPCSUPPORT_H
+
+#include "libmesh/libmesh_config.h"
+
+#ifdef LIBMESH_HAVE_SLEPC
+
+#include "SlepcSupport.h"
+// MOOSE includes
+#include "MultiMooseEnum.h"
+#include "InputParameters.h"
+#include "EigenProblem.h"
+#include "Conversion.h"
+
+namespace Moose
+{
+namespace SlepcSupport
+{
+
+  InputParameters
+  getSlepcValidParams()
+  {
+    InputParameters params = emptyInputParameters();
+
+    // eigen solve type
+    MooseEnum eigen_solve_type("POWER ARNOLDI KRYLOVSCHUR JD");
+    params.addParam<MooseEnum>("eigen_solve_type", eigen_solve_type,
+                               "POWER: Power / Inverse / RQI "
+                               "ARNOLDI: Arnoldi "
+                               "KRYLOVSCHUR: Krylov-Schur "
+                               "JD: Jacobi-Davidson ");
+   return params;
+  }
+
+  InputParameters
+  getSlepcEigenProblemValidParams()
+  {
+    InputParameters params = emptyInputParameters();
+
+    // We are solving a Non-Hermitian eigenvalue problem by default
+    MooseEnum eigen_problem_type("HEP NHEP GHEP GNHEP GHIEP");
+    params.addParam<MooseEnum>("eigen_problem_type", eigen_problem_type,
+                               "HEP: Hermitian "
+                               "NHEP: Non-Hermitian "
+                               "GHEP: Gerneralized Hermitian "
+                               "GNHEP: Generalized Non-Hermitian "
+                               "GHIEP: Generalized indefinite Hermitian ");
+
+
+    // Which eigenvalues are interest
```

Comment:
"Which eigenvalues are we interested in" or "Which eigenvalues are of interest"...